pkgs/stdenv/generic/make-derivation.nix

```
@@ -545,7 +545,12 @@ lib.extendDerivation
        # binaries). By writing this to $out, Nix can find and register
        # them as runtime dependencies (since Nix greps for store paths
        # through $out to find them)
-       args = [ "-c" "export > $out" ];
+       args = [ "-c" ''
+         export > $out
+         for var in $passAsFile; do
+             eval echo "\$(< \$${var}Path)" >> $out
```

Two suggestions basically,
 - `_` to avoid clobbering
 - `${!` as a better meta-variable name dereferencing syntax. (eval is evil)

```suggestion
         for _var in $passAsFile; do
             _fullVarName="${var}Path"
             echo "$(<"${!_fullVarName}")" >> $out
```

I hope I got it right.
We have some tests in `pkgs/test/stdenv/default.nix`. It'd be amazing if you could add one that checks that such a reference exists.

---

> `_` to avoid clobbering

Avoid clobbering what?

---

The global variable namespace, but you're right, that's not necessary. The only relevant variables end in `Path`, so we can take advantage of that.

```suggestion
         for var in $passAsFile; do
             pathVar="${var}Path"
             echo "$(<"${!pathVar}")" >> $out
```

---

Maybe put a newline in front so files can start with `-` without triggering echo flag handling?

```suggestion
         newline='
         '
         for var in $passAsFile; do
             pathVar="${var}Path"
             echo "${newline}$(<"${!pathVar}")" >> $out
```

---

I just switched it out for printf.